### DISCUSSION TITLE
Class override

### DISCUSSION THREAD

**Discussion by marcotrumpet** [vfsfitvnm/frida-il2cpp-bridge#260] [Q&A]
Hi, 
is there a way to intercept a class so that every time someone calls that type I can inject a specific fields in the constructor?

**Comment by vfsfitvnm** [comment]
Nope, Il2Cpp is purely static - but would you elaborate on "intercept a class", "calls that type" and "inject a specific fields in the constructor"?

**Comment by marcotrumpet** [comment]
Sure, so let's say I have
```cs
public class Class1
{
	// Fields
	private int field1;

	// Methods
	public double method1() { }
}
```
`Class 1` is referred by many other methods in other classes and I don't want to override all of those methods. 
I was wondering if there was a way to override `field1` value every time `Class1` constructor is called.

Don't know if it's clear enought

**Reply by vfsfitvnm** [reply]
Ok, now it is! You just have to define a custom implementation for the constructor(s) of that class (the constructor method name is `.ctor`):

```ts
Class1.method(".ctor").implementation = function (...args: any[]) {
  this.method(".ctor").invoke(...args);
  this.field<number>("field1").value = 666;
}
```

**Reply by marcotrumpet** [reply]
So do I need to invoke and then set the new value for the field? Also the args list is like a default parameter or I need to add my own?

**Reply by ChuJiani** [reply]
`...args: any[]` is a TypeScript syntax, which allows you to represent an indefinite number of arguments as an array. Therefore, you don't need to fill in any specific arguments in the `...args: any[]` part, as it will automatically collect all arguments passed to the method into an array.
